### ISSUE TITLE
Feature request: Add a clean way to use transforms when using Cucumber 3

### ISSUE THREAD

**Issue by kratob** [makandra/cucumber_factory#30]
Cucumber_factory has supported Cucumber transforms for a while where all values passed to cucumber_factory were transformed according to all defined global transforms.

This enabled us to do things like 
```
Given there is a user with roles "[projects, accounting]"
```
or similar.

Cucumber 3 dropped support for global transforms. Cucumber factory has then been improved to natively handle integers and arrays, but there is still an occasional need for custom transforms.

This lead to workaround like this one: https://makandracards.com/makandra/72371-cucumber_factory-how-to-keep-using-cucumber-2-transforms-in-cucumber-3/attachments/8782
Basically, it backports the `Transform` function, relying on cucumber_factory to keep using it.

I propose to add this feature to cucumber_factory itself, and provide a `CucumberFactory.register_transform` method:

Usage would be
```
CucumberFactory.register_transform(/file:(\w+)/) do |path|
  File.open(fixture_path.join(path))
end
```

**Comment by triskweline** [comment]
Is assigning file stream the one use case you have for this, or are there others?

This would be straightforward to implement if transforms are only applied to strings (in double-quotes). 

Cucumber Factory can now handle arrays and integers without quotes, but allowing users to hook into the syntax parsing like that would not be trivial.

**Comment by kratob** [comment]
There are others. Current usecase was steps of the form

```
Given there is a task with the duration "3h10m10s"
```

Could obviously been written as `with the duration "11410", but that seemed almost unreadable.

I know our newsroom-apps make also some heavy use out of this.

Restricting it to quoted strings would be fine (and consistent with how transforms worked).

**Comment by makmic** [comment]
We just decided not to invest our resources into this feature. It solves a complex problem for a niche feature which can already be handled following the instructions in [this card](https://makandracards.com/makandra/72371-cucumber_factory-how-to-keep-using-cucumber-2-transforms-in-cucumber-3). 
If there is further interest in this topic, we're open to pull requests.
